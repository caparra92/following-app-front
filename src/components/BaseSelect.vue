<template>
    <ion-row>
    <ion-col>
      <ion-item class="container">
        <select
          v-bind="$attrs"
          class="field field-select"
          @select="$emit('update:modelValue', $event.target.value)"
        >
        <option value="">Select an option...</option>
        <option :value="option.id" v-for="option in modelValue" :key="option.id">{{ option.name }}</option>
        </select>
      </ion-item>
    </ion-col>
  </ion-row>
</template>
<script setup>
import { onMounted } from 'vue';
import { IonRow, IonCol, IonItem, IonIcon } from '@ionic/vue';
onMounted(() => {
  const selectedValue = props.modelValue[0];
});

const props = defineProps({
  label: {
    type: String,
    default: "",
  },
  modelValue: {
    type: [Object, Array, String, Number],
    default: "",
  }
});
</script>
<style scoped>
.container {
  width: 100%;
  padding: 0;
  margin: 0;
}

</style>