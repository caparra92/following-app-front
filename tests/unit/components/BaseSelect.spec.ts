import { mount, VueWrapper } from "@vue/test-utils";
import BaseSelect from "@/components/BaseSelect.vue";

describe('BaseSelect tests', () => {

    let wrapper: VueWrapper;

    beforeEach(() => {
        wrapper = mount(BaseSelect, {
            props: {
                label: 'Category',
                modelValue: ['Capoeira', 'Running', 'Jumping']
            }   
        });
    });

    test('Should match with snapshot', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('Should verify that component exists', () => {
        expect(wrapper.exists()).toBeTruthy();
    });

    test('Should have <ion-item/> tag with class = "icon"', () => {
        const ionItem = wrapper.findComponent('ion-item');
        expect(ionItem.attributes().class).toBe("container");
    });

    test('Should have <select/> tag with classes = ["field", "field-select"]', () => {
        const select = wrapper.find('select');
        expect(select.classes()).toEqual(['field','field-select']);
    });

    test('Should have <select/> tag with props label and modelValue', () => {
        expect(wrapper.props().label).toBe('Category');
        expect(wrapper.props().modelValue).toEqual(['Capoeira', 'Running', 'Jumping']);
    });
});