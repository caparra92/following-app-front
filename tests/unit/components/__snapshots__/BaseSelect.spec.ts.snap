// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseSelect tests Should match with snapshot 1`] = `
<ion-row class="">
  <ion-col class="">
    <ion-item class="container"><select class="field field-select">
        <option value="">Select an option...</option>
        <option></option>
        <option></option>
        <option></option>
      </select></ion-item>
  </ion-col>
</ion-row>
`;
